storage/innobase/fts/fts0fts.cc

```diff
@@ -2715,21 +2718,29 @@ fts_cmp_set_sync_doc_id(
 func_exit:
 
 	if (UNIV_LIKELY(error == DB_SUCCESS)) {
+		if (!created_trx) {
+			return DB_SUCCESS;
+		}
 		fts_sql_commit(trx);
```

Comment:
It seems to me that we had better simply add the lines
```c++
	if (!created_trx) {
		return error;
	}
```
right after `func_exit:`. We do not seem to want any `goto retry` in that case.